lib/table.js

```diff
@@ -167,16 +173,20 @@ module.exports = class Table {
   }
 
   row(opts) {
+    // Defer rendering of the headers until a row is added
+    if (this._rowCount === this._headers.length) {
+      this._pending.push(() => this._renderHeader())
+    }
```

Comment:
I don't mind this getting deferred. However, I think that we now need a unit test for a table that has a header, but no rows, and make sure the header is still rendered. Because this probably needs some additional code to work. Something like check in `_end()` whether the header got already rendered, and if not, execute `this._pending.push(() => this._renderHeader())` there.